### PR TITLE
some CR nits

### PR DESCRIPTION
**Pull request by ryan-williams**

### CODE REVIEW

File: adam-core/src/main/scala/org/bdgenomics/adam/util/MdTag.scala [hunk 1]
```diff
@@ -89,7 +92,7 @@ object MdTag {
             }
             
             if (matchingBases.toInt == usedMatchingBases) {
-              mdTagStringOffset += matchingBases.size
+              mdTagStringOffset += matchingBases.length
```
Comment (arahuja) [inline adam-core/src/main/scala/org/bdgenomics/adam/util/MdTag.scala]: OOC, is there a difference here?

File: adam-core/src/main/scala/org/bdgenomics/adam/util/MdTag.scala [hunk 2]
```diff
@@ -432,7 +444,7 @@ class MdTag(
    */
   def end(): Long = {
     val ends = matches.map(_.end - 1) ::: mismatches.keys.toList ::: deletions.keys.toList
-    ends.reduce(_ max _)
+    ends.max
```
Comment (arahuja) [inline adam-core/src/main/scala/org/bdgenomics/adam/util/MdTag.scala]: Are these the same? Do they treat empty lists the same?

File: adam-core/src/main/scala/org/bdgenomics/adam/util/MdTag.scala (hunk 1 above)
Comment (ryan-williams) [inline adam-core/src/main/scala/org/bdgenomics/adam/util/MdTag.scala]: IntelliJ suggested it, said it is recommended for arrays and strings. Seems like the impl for string just immediately delegates to `.length`, so maybe it's just about saving one call of indirection

File: adam-core/src/main/scala/org/bdgenomics/adam/util/MdTag.scala (hunk 2 above)
Comment (ryan-williams) [inline adam-core/src/main/scala/org/bdgenomics/adam/util/MdTag.scala]: Seems like it:
```
scala> List[Int]().reduce(_ max _)
java.lang.UnsupportedOperationException: empty.reduceLeft
  at scala.collection.LinearSeqOptimized$class.reduceLeft(LinearSeqOptimized.scala:137)
  at scala.collection.immutable.List.reduceLeft(List.scala:84)
  at scala.collection.TraversableOnce$class.reduce(TraversableOnce.scala:206)
  at scala.collection.AbstractTraversable.reduce(Traversable.scala:104)
  ... 33 elided

scala> List[Int]().max
java.lang.UnsupportedOperationException: empty.max
  at scala.collection.TraversableOnce$class.max(TraversableOnce.scala:227)
  at scala.collection.AbstractTraversable.max(Traversable.scala:104)
  ... 33 elided
```

**Pull request merged**